docs/reference/api-conventions.asciidoc

```diff
@@ -195,6 +195,13 @@ request that targets an excluded alias. For example, if `test3` is an index
 alias, the pattern `test*,-test3` still targets the indices for `test3`. To
 avoid this, exclude the concrete indices for the alias instead.
 
+You can also exclude clusters from a list of clusters to search using the `-` character:
+`remote*:*,-remote1:*,-remote4:*` will search all clusters with an alias that starts
+with "remote" except for "remote1" and "remote4". Note that to exclude a cluster
+with this notation you must exclude all of its indexes. Excluding a subset of indexes
+on a remote cluster is currently not supported. For example, this will throw an exception:
```

Comment:
Excluding a subset of indexes is supported but not quite this way? You can do `remote1:-index1`? 

Do we reject `remote*:abc,-remote1:abc`? I am wondering what makes more sense, and also whether this would be perceived as a limitation from Kibana. I guess more of a convention: "if you want to exclude the cluster, use wildcard for indices".

Reply:
> You can do remote1:-index1?

Hmm, I haven't tried that. Good question. I'll look into it.

> Do we reject `remote*:abc,-remote1:abc`? I am wondering what makes more sense

Yes, the current code rejects that.

I decided not to try to tackle that because it is a slippery slope. At this point in the code where it is handled, we don't have any idea what indexes exist on remote clusters. My goal here was to just exclude entire clusters. For your example, it would be fairly straightforward (though more work) to determine that it is a full cluster exclusion. But then a user might expect they can do `remote*:abc*,-remote1:abc` too. But to truly handle that, we'd have to change the sub-search section of TransportSearchAction to send a different request to remote1 then we send to all the other clusters. And the CCS code pathways are already complicated enough that I didn't think that was worth it.

> I guess more of a convention: "if you want to exclude the cluster, use wildcard for indices".

Right, my view is just to have a rule "you can exclude the entire cluster" here by using `*` in the index position.
So we could allow `remote*:abc,-remote1:abc` which is a negation of concrete indexes, but then we have to explain to users why any wildcard other than just `remote1:*` is not allowed.

Reply:
I think your thinking is pragmatic, let's keep it this way.

Reply:
> You can do remote1:-index1?

Thanks for pointing that out. I tested it and it does work in the same way the it does for local indexes. Which has an annoying quirk that you have to use a wildcard in the "inclusion" list to exclude a concrete index.

Example:

This works to exclude `blogs`:

```
POST web_traffic,b*,-blogs/_async_search
```

but this does not:

```
POST web_traffic,blogs,-blogs/_async_search

// fails with
  "error": {
    "type": "status_exception",
    "reason": "error while executing search",
    "caused_by": {
      "type": "index_not_found_exception",
      "reason": "no such index [-blogs] and if you intended to exclude this index, ensure that you use wildcards that include it before explicitly excluding it",
      "resource.type": "index_or_alias",
      "resource.id": "-blogs",
      "index_uuid": "_na_",
      "index": "-blogs"
    }
  }
```

It used to just say "no such index [-blogs]" which is very misleading. [I recently added](https://github.com/elastic/elasticsearch/pull/97866) the extra note "if you intended to exclude this index, ensure that you use wildcards that include it before explicitly excluding it".


So similarly, for remote clusters, this works:

```
POST remote*:blog*,remote1:-blogs/_async_search
```

but this fails with the original misleading error message:

```
POST remote*:blogs,remote1:-blogs/_async_search

  "error": {
    "type": "status_exception",
    "reason": "error while executing search",
    "caused_by": {
      "type": "index_not_found_exception",
      "reason": "no such index [-blogs]",
      "index_uuid": "_na_",
      "resource.type": "index_or_alias",
      "resource.id": "-blogs",
      "index": "-blogs"
    }
  }
```

so I'll file a bug to improve that error message.